Content.Server/Salvage/SalvageSystem.cs

```diff
@@ -0,0 +1,344 @@
+            if (bp == null)
+            {
+                Report("salvage-system-announcement-spawn-debris-disintegrated");
+                return false;
+            }
+            var salvageEntityId = bp.GridEntityId;
+            component.AttachedEntity = salvageEntityId;
+
+            var pulledTransform = EntityManager.GetComponent<TransformComponent>(salvageEntityId);
+            pulledTransform.Coordinates = EntityCoordinates.FromMap(_mapManager, spl);
+            pulledTransform.WorldRotation = spAngle;
+
+            Report("salvage-system-announcement-arrived", ("timeLeft", HoldTime.TotalSeconds));
+            return true;
+        }
+        private void Report(string messageKey) =>
+            _chatManager.DispatchStationAnnouncement(Loc.GetString(messageKey), Loc.GetString("salvage-system-announcement-source"));
+        private void Report(string messageKey, params (string, object)[] args) =>
+            _chatManager.DispatchStationAnnouncement(Loc.GetString(messageKey, args), Loc.GetString("salvage-system-announcement-source"));
+
+        private void Transition(SalvageMagnetComponent magnet, TimeSpan currentTime)
+        {
+            switch (magnet.MagnetState.StateType)
+            {
+                case MagnetStateType.Attaching:
+                    if (SpawnSalvage(magnet))
+                    {
+                        magnet.MagnetState = new MagnetState(MagnetStateType.Holding, currentTime + HoldTime);
+                    }
+                    else
+                    {
+                        magnet.MagnetState = new MagnetState(MagnetStateType.CoolingDown, currentTime + CooldownTime);
+                    }
+                    break;
+                case MagnetStateType.Holding:
+                    Report("salvage-system-announcement-losing", ("timeLeft", DetachingTime.TotalSeconds));
+                    magnet.MagnetState = new MagnetState(MagnetStateType.Detaching, currentTime + DetachingTime);
+                    break;
+                case MagnetStateType.Detaching:
+                    if (magnet.AttachedEntity.HasValue)
+                    {
+                        SafeDeleteSalvage(magnet.AttachedEntity.Value);
+                    }
+                    else
+                    {
+                        Logger.ErrorS("salvage", "Salvage detaching was expecting attached entity but it was null");
+                    }
+                    Report("salvage-system-announcement-lost");
+                    magnet.MagnetState = new MagnetState(MagnetStateType.CoolingDown, currentTime + CooldownTime);
+                    break;
+                case MagnetStateType.CoolingDown:
+                    magnet.MagnetState = MagnetState.Inactive;
+                    break;
+            }
+        }
+
+        public override void Update(float frameTime)
+        {
+            foreach (var gridIdAndState in _salvageGridStates)
+            {
```

Comment:
If salvage stuff is going into a dictionary and not its own component you'll also need to check if it's deleted.

Reply:
Added clearing the data when the round ends to prevent deleted data from being in the salvage state.

Reply:
Don't won't handle all grid deletions though will it? Like if I run rmgrid on it during a round?

Reply:
It is listening for GridRemovalEvent now